src/compiler/binder.ts

```diff
@@ -1851,6 +1854,9 @@ namespace ts {
                 case SyntaxKind.CaseBlock:
                     return ContainerFlags.IsBlockScopedContainer;
 
+                case SyntaxKind.DoExpression:
+                    return ContainerFlags.IsContainer | ContainerFlags.HasLocals | ContainerFlags.IsBlockScopedContainer;

```

Comment:
Hi, I don't quite clear about what those flags mean. Is this appropriate? Should I add `IsControlFlowContainer` when it is `async` do expression which does not start a new `return`/`break`/`continue` context. @ahejlsberg can you help me with this problem?